support-models/src/main/scala/com/gu/support/workers/states/PaymentMethodDisplayFields.scala

```diff
@@ -0,0 +1,98 @@
+package com.gu.support.workers.states
+
+import cats.syntax.functor._
+import com.gu.support.encoding.Codec
+import com.gu.support.encoding.Codec.deriveCodec
+import com.gu.support.workers.states.WireFields.{DirectDebitWirePaymentMethodDisplayFields, PaymentMethodDisplayFieldsWire, JustTypeWirePaymentMethodDisplayFields}
+import io.circe.syntax._
+import io.circe.{Decoder, Encoder, HCursor, Json}
+
+sealed trait PaymentMethodDisplayFields{
+  def `type`: String
+}
+
+sealed trait JustTypePaymentMethodDisplayFields extends PaymentMethodDisplayFields
+
+case object CreditCardDisplayFields extends JustTypePaymentMethodDisplayFields{
+  override val `type` = "CreditCardReferenceTransaction"
+}
+case object PaypalDisplayFields extends JustTypePaymentMethodDisplayFields{
+  override val `type` = "Paypal"
+}
+case class DirectDebitDisplayFields(
+                                          bankAccountNumberMask: String,
+                                          bankSortCode: String,
+                                          bankAccountName: String,
+                                          mandateId: Option[String],
+                                        ) extends PaymentMethodDisplayFields {
+  val `type`: String = "BankTransfer"
+
+}
+
+object WireFields {
```

Comment:
I used the intermediate wire fields to get the json serialisation to work the way I wanted. Maybe there is a more elegant  way to do this in circe without them.

Reply:
Why do you need the intermediate classes? They look identical

Reply:
for non direct debit payment methods we just have an object that tells us what type of payment it is, but we need this to serialise to a json object with a type field.  
The "wire" case classes resemble the desired json structure more directly, So there is a direct debit one and a justType one that contains a type field.
We could remove some duplication if we unified the two direct debit classes (making the same class extend both traits) but I though it was cleaner this way

Having said all of that, there is probably a way to make circe generate this for us without intermediate classes, I just don't know if it would be easier to understand than this